### PR TITLE
Add PHP 7.0 - 7.3 on Travis CI

### PR DESCRIPTION
**Pull request by sodabrew**
Also switch distribution from Trusty to Xenial as mentioned here https://github.com/travis-ci/travis-ci/issues/9717#issuecomment-421016620

### CODE REVIEW

**Comment by sodabrew** [comment]
Hm, I assumed due to presence of a .travis.yml that https://travis-ci.org/php/pecl-networking-ssh2 would be an active build. Is this something an admin would like to enable? Are you building this code on a different build/test service?

**Comment by sodabrew** [comment]
CC @langemeijer @remicollet ☝️

**Comment by mhagstrand** [comment]
I added the `.travis.yml` a while back in this PR: https://github.com/php/pecl-networking-ssh2/pull/25. I was hoping it would get enabled.  I figured it would make development a little easier.

**Comment by sodabrew** [comment]
@langemeijer merged the Travis CI config file but didn't enable Travis CI for the repo, which isn't a useful combination. Strongly request enabling the service.

**Comment by nikic** [comment]
I've enabled Travis for this repository. Will leave merging this to a maintainer...

**Comment by sodabrew** [comment]
Tests passing on all versions of PHP 7.x!

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 sudo: required
```
Comment (sergeyklay) [inline .travis.yml]:
```diff
- sudo: required
```

Travis is deprecating the `sudo` keyword and moves everything to the same infrastructure (`sudo` really selects between two infrastructures).

See: https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
-dist: trusty
+dist: xenial
```
Comment (sergeyklay) [inline .travis.yml]:
```diff
- dist: xenial
```

I think you can remove `dist` at all

**Comment by andypost** [comment]
Is there a way to create new release? Trying to package extension to Alpinelinux without success https://github.com/alpinelinux/aports/pull/6226

**Comment by frederikbosch** [comment]
I would also be interested in a new release. Is there anything anyone, including myself, can do to make it happen?

**Comment by andypost** [comment]
And it still require https://github.com/php/pecl-networking-ssh2/pull/36 for 7.3 compatibility

File: .travis.yml (hunk 2 above)
Comment (andypost) [reply .travis.yml]: xenial is default now https://docs.travis-ci.com/user/reference/overview/#linux

Comment (andypost) [reply .travis.yml]: xenial has no 7.3 support on travis https://docs.travis-ci.com/user/reference/xenial/#php-support

So it should switch to `bionic` and drop 7.0 https://docs.travis-ci.com/user/reference/bionic/#php-support

Comment (glensc) [reply .travis.yml]: you **can** use different dist per versions. this example uses different dist for 5.3:
- https://github.com/eventum/rpc/blob/v4.3.0/.travis.yml

Comment (glensc) [reply .travis.yml]: however, you linked doc says:
> The following PHP versions are preinstalled

it doesn't mean the other versions are not supported, they just are downloaded on demand. at least I've seen downloading x.y version messages in build logs.

**Comment by glensc** [comment]
btw, can someone update pecl `Homepage` property to point to this repo instead of [libssh2 homepage](http://www.libssh2.org/)?

- https://pecl.php.net/package/ssh2

**Comment by glensc** [comment]
also, can someone push tags for created releases? so far only some legacy tags are present:
- `RELEASE_0_10`
- `RELEASE_0_11_0`
- `RELEASE_0_11_1`
- `RELEASE_0_11_2`
- `RELEASE_0_11_3`
- `RELEASE_0_12`
- `RELEASE_0_4`
- `RELEASE_0_4_1`
- `RELEASE_0_5`
- `RELEASE_0_6`
- `RELEASE_0_7`
- `RELEASE_0_8`
- `RELEASE_0_9`
- `RELEASE_1_0_4`

if needed I can dig up the commit hashes for someone to create lightweight tags.

as I'm trying to build this for php 7.3, and apparently applying only https://github.com/php/pecl-networking-ssh2/pull/36 is not enough, and not known what is the baseline from the last release as git tags are missing from the repository.

**Pull request closed without merging**